sdk/core/System.ClientModel/tests/Pipeline/HttpClientPipelineTransportTests.cs

```diff
@@ -138,6 +139,364 @@ public async Task SettingResponseContentStreamPreservesHeaders(string headerName
+    {
+        MockHttpClientHandler mockHandler = new(
+            httpRequestMessage =>
+            {
+                return Task.FromResult(new HttpResponseMessage(HttpStatusCode.OK)
+                {
+                    Content = new StringContent("Mock content")
+                });
+            });
+
+        HttpClientPipelineTransport transport = new(new HttpClient(mockHandler));
+
+        using PipelineMessage message = transport.CreateMessage();
+        message.Request.Method = "GET";
+        message.Request.Uri = new Uri("http://example.com");
+
+        await transport.ProcessSyncOrAsync(message, IsAsync);
+
+        Assert.AreEqual("Mock content", message.Response!.Content.ToString());
+    }
+
+    [Test]
+    public async Task BufferedResponseContentEmptyWhenNoResponseContent()
+    {
+        MockHttpClientHandler mockHandler = new(
+            httpRequestMessage =>
+                Task.FromResult(new HttpResponseMessage(HttpStatusCode.OK)));
+
+        HttpClientPipelineTransport transport = new(new HttpClient(mockHandler));
+
+        using PipelineMessage message = transport.CreateMessage();
+        message.Request.Method = "GET";
+        message.Request.Uri = new Uri("http://example.com");
+
+        await transport.ProcessSyncOrAsync(message, IsAsync);
+
+        Assert.AreEqual(0, message.Response!.Content.ToMemory().Length);
+    }
+
+    [Test]
+    public async Task BufferedResponseContentStreamContainsResponseContent()
+    {
+        MockHttpClientHandler mockHandler = new(
+            httpRequestMessage =>
+            {
+                return Task.FromResult(new HttpResponseMessage(HttpStatusCode.OK)
+                {
+                    Content = new StringContent("Mock content")
+                });
+            });
+
+        HttpClientPipelineTransport transport = new(new HttpClient(mockHandler));
+
+        using PipelineMessage message = transport.CreateMessage();
+        message.Request.Method = "GET";
+        message.Request.Uri = new Uri("http://example.com");
+
+        await transport.ProcessSyncOrAsync(message, IsAsync);
+
+        Assert.AreEqual("Mock content", BinaryData.FromStream(message.Response!.ContentStream!).ToString());
+    }
+
+    [Test]
+    public async Task BufferedResponseContentStreamEmptyWhenNoResponseContent()
+    {
+        MockHttpClientHandler mockHandler = new(
+            httpRequestMessage =>
+                Task.FromResult(new HttpResponseMessage(HttpStatusCode.OK)));
+
+        HttpClientPipelineTransport transport = new(new HttpClient(mockHandler));
+
+        using PipelineMessage message = transport.CreateMessage();
+        message.Request.Method = "GET";
+        message.Request.Uri = new Uri("http://example.com");
+
+        await transport.ProcessSyncOrAsync(message, IsAsync);
+
+        Assert.AreEqual(0, message.Response!.ContentStream!.Length);
+    }
+
+    [Test]
+    public async Task BufferedResponseContentStreamCanBeReadMultipleTimes()
```

Comment:
nit: some of these tests seem to only address behavior of `HttpClientTransportResponse` that is agnostic to the transport. Should those tests be simplified and moved to `HttpClientTransportResponseTests`?

Reply:
Interesting.  The response implementation is specific to a transport -- note that the response type itself is a private nested class in the transport.  But I see your point that some of the tests in this file are testing transport.Process and other public methods on transport, whereas others are testing the response implementation.  If you feel strongly about this, I can create a separate class that contains tests specific to the transport's response implementation.

Reply:
> note that the response type itself is a private nested class in the transport.

I hadn't noticed that - makes sense to me then 😄